src/core/vector/qgsvectorlayerutils.h

```diff
@@ -374,17 +374,29 @@ class CORE_EXPORT QgsVectorLayerUtils
     static bool impactsCascadeFeatures( const QgsVectorLayer *layer, const QgsFeatureIds &fids, const QgsProject *project, QgsDuplicateFeatureContext &context SIP_OUT, QgsVectorLayerUtils::CascadedFeatureFlags flags = QgsVectorLayerUtils::CascadedFeatureFlags() );
 
     /**
-     * Given a set of \a fields, attempts to pick the "most useful" field
+     * Given a set of fields, attempts to pick the "most useful" field
      * for user-friendly identification of features.
      *
      * For instance, if a field called "name" is present, this will be returned.
      *
      * Assumes that the user has organized the data with the more "interesting" field
      * names first. As such, "name" would be selected before "oldname", "othername", etc.
      *
+     * If no friendly identifier is found, the function will fallback to the
+     * first available.
+     *
+     * An optional boolean parameter can be used to determine whether the returned
+     * field name is a friendly identifier or not.
+     *
+     * \param fields list of fields to pick a friendly identifier from
+     * \param foundFriendly set to TRUE if the returned field name is a friendly identifier (since QGIS 3.22)
+     * \returns field name
      * \since QGIS 3.18
      */
+#ifdef SIP_RUN
     static QString guessFriendlyIdentifierField( const QgsFields &fields );
+#endif
+    static QString guessFriendlyIdentifierField( const QgsFields &fields, bool *foundFriendly SIP_OUT = nullptr ) SIP_PYNAME( guessFriendlyIdentifierFieldV2 );
```

Comment:
This doesn't quite work -- you've gotta copy the docs too. Otherwise either doxygen and/or sipify get messed up

Reply:
Ahh you mean copy dox within the #ifdef SIP_RUN?

Reply:
yep